Create topology for I2S ALC5682 and ALC1019

udaymb opened the pull request:
Topology to support ALC5682 and ALC1019 codec

plbossart commented on tools/topology/topology2/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,283 @@
+			id	0
+			mclk_freq	24000000
+			bclk_freq	4800000
+			tdm_slot_width	32
```
that seems wrong. this would give you a 150kHz frame rate.

udaymb replied:
updated the changes - referring to
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-jsl-rt5682.m4#L123-L135
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-jsl-rt5682.m4#L176-L181

plbossart commented on tools/topology/topology2/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,283 @@
+			id	0
+			mclk_freq	24000000
+			bclk_freq	4800000
+			tdm_slot_width	32
```
same here, when you have a 24 MHz clock you have to use multiple of 25.

udaymb replied:
updated the changes - referring to
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-jsl-rt5682.m4#L118-L121
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/platform/intel/jsl-rt1015.m4#L11-L18

plbossart commented on tools/topology/topology2/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,283 @@
+# List of all DAIs
+#
+# PCM0 ----> volume -----> SSP1  (Speaker - ALC1019)
+# PCM1 <---> volume <----> SSP0  (Headset - ALC5682)
```
why did you invert PCM and SSPs

Why not use PCM0 for SSP0 and PCM1 for SSP1.

Simple.always.wins

udaymb replied:
Based on previous topologies for JSL and TGL, SSP1 is used for the SPEAKER and SSP0 for headset
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-jsl-rt5682.m4#L26-L27
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-jsl-rt5682.m4#L168
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-tgl-max98357a-rt5682.m4#L77-L80
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-tgl-max98373-rt5682.m4#L27-L32

bardliao commented:
@udaymb Please put the conf file in the tools/topology/topology2/cavs/ folder.

udaymb commented:
> 

As suggested moved to tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf

lgirdwood commented:
@ranj063 not seeing you on the review, can you add yourself to CODEOWNERS for topology2 ?

udaymb commented:
> Pls also make sure the DAI link id matches with machine driver, spk link is created after DMIC in sof_rt5682.c.

DAI link ids are updated as per the machine driver

> Should we leave formats field to default? otherwise it can only play S32_LE.

Removed the entries to use the default

udaymb commented:
After discussion, moved to passthrough instead of volume playback/capture

plbossart commented on tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,250 @@
+# List of all DAIs
+#
+# PCM0 ----> volume -----> SSP1  (Speaker - ALC1019)
+# PCM1 <---> volume <----> SSP0  (Headset - ALC5682)
```
why invert? Please use this more straightforward layout:

PCM0 <---> volume <----> SSP0  (Headset - ALC5682)
PCM1 ----> volume -----> SSP1  (Speaker - ALC1019)

udaymb replied:
Sorry for the mistake, Updated with the changes

plbossart commented on tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,250 @@
+
+		Object.Base.hw_config."SSP0" {
+			id	0
+			mclk_freq	24000000
```
is this real? we haven't use 24MHz since CNL?

udaymb replied:
Made relevant changes, please review

plbossart commented on tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,250 @@
+		Object.Base.hw_config."SSP1" {
+			id	0
+			mclk_freq	24000000
+			bclk_freq	3072000
```
that means mclk and bclk use a different source. I thought there were some changes added in TGL and follow-ups, does this work?

plbossart replied:
??

udaymb replied:
Updating all settings similar to rt1011

https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-tgl-max98357a-rt5682.m4#L380-L385

plbossart reviewed (changes requested):
pipelines can be simplified and mclk settings are very very odd.

udaymb commented:
> The id is unique, since SSP0's id is 1, so DMIC id cannot be 1 again.
Done

> A more friendly name than PortN?
Done

bardliao commented on tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,250 @@
+#
+# PCM0 ----> SSP0  (Headset - ALC5682)
+# PCM1 <---> SSP1  (Speaker - ALC1019)
+# PCM2 <---- DMIC01 (dmic 48k capture)
```
nitpick: Headset is bidirectional and speaker is playback only. So it is better
```
# PCM0 <---> SSP0  (Headset - ALC5682)
# PCM1 ----> SSP1  (Speaker - ALC1019)
```

plbossart replied:
agree with Bard!

udaymb replied:
Sorry a mistake, corrected it

bardliao commented on tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,250 @@
+<include/common/data.conf>
+<include/dais/hw_config.conf>
+<include/pipelines/cavs/passthrough-playback.conf>
+<include/pipelines/cavs/passthrough-capture.conf>
```
Please add `<cavs/common_definitions.conf>` here

udaymb replied:
included

bardliao commented on tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,250 @@
+
+# Pipeline ID:1 PCM ID: 0
+Object.Pipeline {
+    passthrough-playback.1 {
```
Please use `tab` instead of `space` for indentation.

udaymb replied:
updated

bardliao commented:
@udaymb Please add "cavs-rt5682-rt1019\;cavs-rt5682-rt1019.conf" into tools/topology/topology2/cavs/CMakeLists.txt so that we can build this topology with ./scripts/build-tools.sh

plbossart commented on tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf:
```diff
@@ -0,0 +1,250 @@
+		fir_passband_end	 20000
+		fir_stopband_start	 24000
+		fir_passpand_ripple	 10
+		fir_stopband_ripple	 -95000
```
@lgirdwood @juimonen what is the plan for DMIC support? Are we really going to generate a blob in the kernel based on this topology, or is the firmware going to generate the relevant register settings based on topology indications?

udaymb commented:
> @udaymb Please add "cavs-rt5682-rt1019;cavs-rt5682-rt1019.conf" into tools/topology/topology2/cavs/CMakeLists.txt so that we can build this topology with ./scripts/build-tools.sh
Added it in CMakeLists.txt

https://github.com/thesofproject/sof/pull/5231#commitcomment-64270686
Based on this PRs comment, removed dai_index to inherit from parent DAI object

udaymb commented:
https://github.com/thesofproject/sof/pull/5187#discussion_r792989721
this is modified now

https://github.com/thesofproject/sof/pull/5187#discussion_r792989135
modified

**Setting these values for ALC5682**
# SSP0
MCLK 19.2MHz
DAI_CONFIG(SSP, 0, 0, SSP0-Codec,
SSP_CONFIG(I2S, SSP_CLOCK(mclk, 19200000, codec_mclk_in),
SSP_CLOCK(bclk, 2400000, codec_slave),
SSP_CLOCK(fsync, 48000, codec_slave),
SSP_TDM(2, 25, 3, 3),
SSP_CONFIG_DATA(SSP, 0, 24, 0, 0, 0, SSP_CC_BCLK_ES)))

**Setting these values for ALC1019**
# SSP1 DAI FMT s32le
MCLK 19.2MHz
SSP_CONFIG(I2S, SSP_CLOCK(mclk, 19200000, codec_mclk_in),
SSP_CLOCK(bclk, 3072000, codec_slave),
SSP_CLOCK(fsync, 48000, codec_slave),
SSP_TDM(2, 32, 3, 3),
SSP_CONFIG_DATA(SSP, 1, 24))

plbossart reviewed (changes requested):
comments not addressed.

udaymb commented:
>> https://github.com/thesofproject/sof/pull/5187#discussion_r793757912

Updating all settings similar to rt1011
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-tgl-max98357a-rt5682.m4#L380-L385

sys-pt1s commented:
Can one of the admins verify this patch?

lgirdwood commented:
@udaymb ping ?

udaymb commented:
Apologies for the delay - board is still not available for verification
Need to wait till get it

kv2019i commented:
@udaymb I'm turning this to a draft as this is stuck on hw availability.

kv2019i commented:
Stale PR,closing
